ssl/t1_lib.c

```diff
@@ -691,13 +691,13 @@ int tls1_set_groups(uint16_t **pext, size_t *pextlen,
     return 0;
 }
 
-/* TODO(3.0): An arbitrary amount for now. Take another look at this */
-# define MAX_GROUPLIST   40
+# define GROUPLIST_INCREMENT   40
 # define GROUP_NAME_BUFFER_LENGTH 64
 typedef struct {
     SSL_CTX *ctx;
     size_t gidcnt;
-    uint16_t gid_arr[MAX_GROUPLIST];
+    size_t gidmax;
+    uint16_t *gid_arr;
```

Comment:
Why not a stack of gids?

Reply:
It could be done that way in the cb, but it ultimately needs to end up in an array in `tls1_set_groups_list` which is why its done this way.